src/averageAges.js

```diff
@@ -55,7 +65,18 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childFiltered = people.filter(child => onlyWithSon
+    ? child.sex === 'm'
+      && people.some(mother => mother.name === child.mother)
+    : people.some(mother => mother.name === child.mother));
```

Comment:
```suggestion
  const childFiltered = people.filter(child => onlyWithSon
    ? child.sex === 'm' && people.some(
         mother => mother.name === child.mother
       )
    : people.some(mother => mother.name === child.mother)
  );
```
pay attention to the code formatting